LoopKitUI/Views/BasalRateScheduleEditor.swift

```diff
@@ -0,0 +1,132 @@
+            absoluteBounds: supportedBasalRates.first!...supportedBasalRates.last!,
+            recommendedBounds: supportedBasalRates.dropFirst().first!...supportedBasalRates.dropLast().last!,
+            unit: .internationalUnitsPerHour
+        )
+        self.maximumScheduleEntryCount = maximumScheduleEntryCount
+        self.syncSchedule = syncSchedule
+        self.save = save
+    }
+
+    public var body: some View {
+        QuantityScheduleEditor(
+            title: Text("Basal Rates", comment: "Title of basal rate settings page"),
+            description: description,
+            schedule: schedule,
+            unit: .internationalUnitsPerHour,
+            selectableValues: supportedBasalRates,
+            guardrail: guardrail,
+            quantitySelectionMode: .fractional,
+            defaultFirstScheduleItemValue: guardrail.absoluteBounds.lowerBound,
+            scheduleItemLimit: maximumScheduleEntryCount,
+            confirmationAlertContent: confirmationAlertContent,
+            guardrailWarning: BasalRateGuardrailWarning.init(crossedThresholds:),
+            onSave: .asynchronous { quantitySchedule, completion in
+                self.syncSchedule(quantitySchedule.items) { result in
+                    switch result {
+                    case .success(let syncedSchedule):
+                        DispatchQueue.main.async {
+                            self.save(syncedSchedule)
+                        }
+                        completion(nil)
+                    case .failure(let error):
+                        completion(error)
+                    }
+
+                }
+            }
+        )
+    }
+
+    private var description: Text {
+        Text("Your basal rate of insulin is the number of units per hour that you want to use to cover your background insulin needs.", comment: "Basal rate setting description")
+    }
+
+    private var confirmationAlertContent: AlertContent {
+        AlertContent(
+            title: Text("Save Basal Rates?", comment: "Alert title for confirming basal rates outside the recommended range"),
+            message: Text("One or more of the values you have entered are outside of what Tidepool generally recommends.", comment: "Alert message for confirming basal rates outside the recommended range")
+        )
+    }
+}
+
+private struct BasalRateGuardrailWarning: View {
+    var crossedThresholds: [SafetyClassification.Threshold]
+
+    var body: some View {
+        assert(!crossedThresholds.isEmpty)
+        return GuardrailWarning(
+            title: crossedThresholds.count == 1 ? singularWarningTitle(for: crossedThresholds.first!) : multipleWarningTitle,
+            thresholds: crossedThresholds,
+            customCaption: { crossedThresholds in
+                crossedThresholds.allSatisfy({ $0 == .minimum })
+                    ? Text("A setting of 0 U/hr means you will not receive basal insulin or automation.", comment: "Warning text for basal rate of 0 U/hr")
```

Comment:
(Grabbed this copy from the Jira ticket, but I don't think it's accurate. You can still be high-temped with a basal setting of 0 U/hr, so you're still "receiving automation". Discussing now, will likely be updated.)